RoomLocator/RoomLocator.Data/Services/ScadadataService.cs

```diff
@@ -0,0 +1,128 @@
+
+
+
+        private readonly IHttpClientFactory _clientFactory;
+
+        public ScadadataService(IHttpClientFactory clientFactory)
+        {
+            _clientFactory = clientFactory;
+        }
+
+        public async Task<IEnumerable<ScadadataViewModel>> GetSensorData()
+        {
+            const string url = "https://scadadataapi.azurewebsites.net/api/values";
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            //var mc = new  ScadadataService();
+
+
+            var client = _clientFactory.CreateClient("dtu-cas");
+            var response = await client.SendAsync(request);
+
+            return !response.IsSuccessStatusCode ? null : JsonConvert.DeserializeObject<List<ScadadataViewModel>>(await
+                response.Content.ReadAsStringAsync());
+        }
+        
+        public async Task<IEnumerable<ScadadataScoresModel>> GetListOfScores()
+        {
+            var sd = await GetSensorData();
+            var scadadataViewModels = sd.ToList();
+            var temperatures = scadadataViewModels.Where(item => item.Type == "Temperature").ToList();
+            var lights = scadadataViewModels.Where(item => item.Type == "Light").ToList(); 
+            var sounds = scadadataViewModels.Where(item => item.Type == "Sound").ToList(); 
+            var availableSeats = scadadataViewModels.Where(item => item.Type == "available seats").ToList(); 
+            var maxAvailableSeats = scadadataViewModels.Where(item => item.Type == "max available seats").ToList();
+            var list = new List<ScadadataScoresModel>
+            {
+                new ScadadataScoresModel("Temperature",
+                    1 - FindAverageDeviation(temperatures, TemperatureOptimal, TemperatureMaxVariance)),
+                new ScadadataScoresModel("Sound",
+                    1-FindAverageDeviation(sounds, SoundOptimal, SoundMaxVariance)),
+                new ScadadataScoresModel("Light",
+                    1-FindAverageDeviation(lights, LightOptimal, LightMaxVariance)),
+                new ScadadataScoresModel("Seats Available",
+                    1-FindAverageDeviation(availableSeats, maxAvailableSeats[0].Value, 
+                        maxAvailableSeats[0].Value-40))
+            };
+            return list;
+
+
+        }
+        public async Task<double> GetWeightedScore()
+        {
+            var sd = await GetSensorData();
+            var scadadataViewModels = sd.ToList();
+            var temperatures = scadadataViewModels.Where(item => item.Type == "Temperature").ToList();
+            var lights = scadadataViewModels.Where(item => item.Type == "Light").ToList(); 
+            var sounds = scadadataViewModels.Where(item => item.Type == "Sound").ToList(); 
+            var availableSeats = scadadataViewModels.Where(item => item.Type == "available seats").ToList(); 
+            var maxAvailableSeats = scadadataViewModels.Where(item => item.Type == "max available seats").ToList();
```

Comment:
Alt det her er duplikeret kode. Mon ikke `GetSensorData()` burde returnere data i det format du skal bruge det i i stedet for? Eventuelt ret din ViewModel så den passer det format i stedet for. Eller lav en ny ViewModel og brug AutoMapper til at mappe til det rigtige format.